### PR TITLE
[QI2-896] Added (auto)member_id fetching functionality

### PR DESCRIPTION
**Pull request by NischalQuTech**

### CODE REVIEW

File: quantuminspire/util/configuration.py [hunk 1]
```diff
@@ -155,10 +156,43 @@ def store_tokens(self, host: Url, tokens: TokenInfo) -> None:
...
+
+        :param host: the hostname of the api for which the tokens are intended
+        :param access_token: OAuth access token for the api
+        :return: int
+
+        This function fetches the member_id to be used for creating a project.
+        """
+
+        url_members = f"{host}/members"
+        response = requests.get(url_members, headers={"Authorization": "Bearer " + access_token})
```
Comment (oschusler) [inline quantuminspire/util/configuration.py]: Shouldn't we do this via the compute-api-client?

File: quantuminspire/util/configuration.py [hunk 2]
```diff
@@ -155,10 +156,43 @@ def store_tokens(self, host: Url, tokens: TokenInfo) -> None:
...
+        members: List[Dict[str, Union[int, Any]]] = response.json()
+        if len(members) == 1:
+            print(f"Using member ID {members[0]['id']}")
+            return members[0]["id"]
+
+        member_ids = [member["id"] for member in members]
+        print("Choose a member ID from the list for project configuration.")
```
Comment (oschusler) [inline quantuminspire/util/configuration.py]: Can't we use typer for this? It's what it's for, right?

File: quantuminspire/util/configuration.py (hunk 1 above)
Comment (NischalQuTech) [reply quantuminspire/util/configuration.py]: I changed it to using compute api client now.

File: quantuminspire/util/configuration.py (hunk 2 above)
Comment (NischalQuTech) [reply quantuminspire/util/configuration.py]: Replaced print statements with typer.echo

File: quantuminspire/util/api/remote_backend.py [hunk 3]
```diff
@@ -95,6 +98,31 @@ async def _create_flow(
...
+
+        typer.echo("Choose a member ID from the list for project configuration.")
+        json_string = "[" + ",".join(member.model_dump_json(indent=4) for member in members) + "]"
+        typer.echo(json_string)
+
+        member_ids = [member.id for member in members]
+
+        while True:
+            try:
+                member_id = int(input("Please enter an integer id: "))
```
Comment (eliasKA) [inline quantuminspire/util/api/remote_backend.py]:
```suggestion
                member_id = int(input("Please enter one of the given ids: "))
```

**Review by eliasKA — APPROVED** [review]
LGTM ✅,  left one small suggestion but nothing blocking

File: quantuminspire/util/api/remote_backend.py [hunk 4]
```diff
@@ -86,6 +88,7 @@ async def _create_flow(
     ) -> int:
         """Call the necessary methods in the correct order, with the correct parameters."""
         async with ApiClient(self._configuration) as api_client:
+            await self._set_team_member_id(api_client)
```
Comment (oschusler) [inline quantuminspire/util/api/remote_backend.py]: This is only when running an algorithm, not when you log in, right?

Comment (NischalQuTech) [reply quantuminspire/util/api/remote_backend.py]: Moved it to login logic

**Review by oschusler — APPROVED** [review]

**Pull request merged**